tools/run_defacing_script.pl

```diff
@@ -0,0 +1,687 @@
+    my $deface_ref = $deface_dir . '/' . basename($$ref_file{File});
+    $deface_ref    =~ s/\.mnc$/_defaced\.mnc/;
+
+    # create a hash with all information about the defaced images and add the
+    # reference t1 defaced image to it
+    my %defaced_images;
+    # append the reference t1 defaced image to the hash
+    $defaced_images{$deface_ref}{InputFileID} = $$ref_file{FileID};
+    $defaced_images{$deface_ref}{Scan_type}   = $$ref_file{Scan_type};
+
+    # for each files in $session_files, append the defaced images to the hash
+    foreach my $scan_type (keys $session_files) {
+        my %files = %{ $$session_files{$scan_type} };
+        foreach my $fileID (keys %files) {
+            my $deface_file   = $deface_dir . '/' . basename($files{$fileID});
+            $deface_file      =~ s/\.mnc$/_defaced\.mnc/;
+            $defaced_images{$deface_file}{InputFileID} = $fileID;
+            $defaced_images{$deface_file}{Scan_type}    = $scan_type;
+        }
+    }
+
+    # make sure all the files can be found on the filesystem
+    foreach my $file (keys %defaced_images) {
+        return undef unless (-e $file);
+    }
+
+    return %defaced_images;
+}
+
+
+=pod
+
+=head3 register_defaced_files($defaced_images)
+
+Registers the defaced images using C<register_processed_data.pl>.
+
+INPUT: hash with the defaced images storing their input FileID and scan type
+
+=cut
+
+sub register_defaced_files {
+    my ($defaced_images) = @_;
+
+    my $register_cmd = "register_processed_data.pl "
+                       . " -profile $profile "
+                       . " -sourcePipeline MINC_deface "
+                       . " -tool 'uploadNeuroDB/bin/deface_minipipe.pl' "
+                       . " -pipelineDate $today "
+                       . " -coordinateSpace native "
+                       . " -outputType defaced ";
+
+    foreach my $file (keys $defaced_images) {
+        my $input_fileID = $$defaced_images{$file}{InputFileID};
+        my $scan_type    = $$defaced_images{$file}{Scan_type} . "-defaced";
+
+        # verify that the scan type exists in mri_scan_type, if not create it
+        create_defaced_scan_type($scan_type);
+
+        # append file, scan type, sourceFileID & inputFileIDs to the command
+        $register_cmd .= " -inputFileIDs $input_fileID "
+                         . " -sourceFileID $input_fileID"
+                         . " -scanType $scan_type "
+                         . " -file $file";
+
+        # register the scan in the DB
+        system($register_cmd);
+    }
+}
+
+
+=pod
+
+=head3 create_defaced_scan_type($scan_type)
+
+Function that inserts a new scan type in C<mri_scan_type> if the scan type does not
+already exists in C<mri_scan_type>.
+
+INPUT: the scan type to look for or insert in the C<mri_scan_type> table
+
+=cut
+
+sub create_defaced_scan_type {
+    my ($scan_type) = @_;
+
+    my $select     = "SELECT COUNT(*) FROM mri_scan_type WHERE Scan_type = ?";
```

Comment:
We could also simply do

```perl
my $insert = 'INSERT IGNORE INTO mri_scan_type SET Scan_type = ?';
$dbh->prepare($insert)->execute($scan_type);
```

Reply:
I am not sure INSERT IGNORE would work in this case. Here, before creating a new scan type, I just want to make sure it did not already exist (note that Scan_type is not a UNIQUE entry so if I use your suggestion, you would end up with multiple t1-deface entries instead of just 1)

Reply:
You are right. I wonder why column `Scan_type` is not `UNIQUE` though. I think that would make sense.